.github/workflows/ci-validation-duration.yaml

```diff
@@ -0,0 +1,34 @@
+name: Measure Validation Test Duration
+
+on:
+  workflow_dispatch:
+  # run these jobs against the PR + master.  To run tests against just the PR, use "push" instead of "pull_request"
+  pull_request:
+    # run these jobs when a PR is opened, edited, reopened, or updated (synchronize)
+    # edited = title, body, or the base branch of the PR is modified
+    # synchronize = commit(s) pushed to the pull request
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
```

Comment:
I suggest making this a workflow_dispatch only. Then you can run it once against your merge base commit (usually main) and once against your test commit and compare the output.

Another option is to run this on label, and add a custom label that you can use to trigger the action on any PR. It's only marginally easier than a manual trigger through the actions UI, so I don't think it's worth the overhead to get that set up.

```suggestion
```